### PR TITLE
Removes xref to missing file and fixes broken links

### PR DESCRIPTION
**Pull request by ryandeussing**
Both of these commits are addressing validation errors caught by the web-based tool that prepares titles for the Red Hat Customer Portal and which block publication.

It's not ideal to catch these problems at the final stage of pre-publication. A command-line validator is in development that may allow writers themselves to catch and fix these issues far in advance of future releases.

### CODE REVIEW

**Comment by lzap** [comment]
Yeah, in this repo we have a validator available. It runs automatically for the master branch, for stable branches we have decided to turn it off because RHEL links gets broken so often that maintaining them in all branches was painful.

You can easily run them locally, follow our readme on setting up environment on your laptop:

```
dnf -y install rubygem-asciidoctor rubygem-asciidoctor-pdf make linkchecker
cd guides
make clean
make html linkchecker
```

You almost fixed all of them, there is one more issue, would you mind fixing it too?

```
Start checking at 2021-09-22 18:04:42+002
18 threads active,     1 link queued,    1 link in  20 URLs checked, runtime 1 seconds
	32 threads active,    34 links queued,   62 links in 128 URLs checked, runtime 6 seconds

URL        `https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/indexsec-using_firewalls#sec-Getting_started_with_firewalldchap-federal_standards_and_regulations#sec-Enabling-FIPS-Mode'
Name       `Enabling FIPS Mode'
Parent URL file:///home/lzap/Projekty/foreman-documentation/guides/build/Installing_Proxy_on_Debian/index-foreman-el.html, line 115, col 27
Real URL   https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/indexsec-using_firewalls#sec-Getting_started_with_firewalldchap-federal_standards_and_regulations%23sec-Enabling-FIPS-Mode
Check time 5.004 seconds
Size       17.48KB
Result     Error: 404 Not Found
32 threads active,    65 links queued,  113 links in 215 URLs checked, runtime 11 seconds
32 threads active,    35 links queued,  143 links in 220 URLs checked, runtime 16 seconds
24 threads active,     0 links queued,  186 links in 224 URLs checked, runtime 21 seconds
 5 threads active,     0 links queued,  205 links in 227 URLs checked, runtime 26 seconds

Statistics:
Downloaded: 2.21MB.
Content types: 36 image, 141 text, 0 video, 0 audio, 9 application, 0 mail and 24 other.
URL lengths: min=16, max=258, avg=91.

That's it. 210 links in 227 URLs checked. 0 warnings found. 1 error found.
```

File: guides/doc-Managing_Hosts/topics/Registering_Hosts.adoc [hunk 1]
```diff
@@ -22,7 +22,6 @@ You can also register Atomic Hosts to {ProjectServer} or {SmartProxyServer}.
 
 Use one of the following procedures to register a host:
 
-* xref:registering-a-host-to-project-using-the-global-registration-template_managing-hosts[]
```
Comment (lzap) [inline guides/doc-Managing_Hosts/topics/Registering_Hosts.adoc:25]: Looks like there are multiple occurrences, did you mean to remove them all?

```
[lzap@x1 guides]$ ag "registering-a-host-to-project-using-the-global-registration-template_managing-hosts"
doc-Content_Management_Guide/topics/QuickStart.adoc
154:Use your activation key to register a host to the Content View. For more information on registering a host, see {ManagingHostsDocUR}registering-a-host-to-project-using-the-global-registration-template_managing-hosts

doc-Managing_Hosts/topics/using_insights_with_satellite_hosts.adoc
12:For more information, see {ManagingHostsDocURL}registering-a-host-to-project-using-the-global-registration-template_managing-hosts[Registering a Host to {ProjectName} Using the Global Registration Template] in the _Managing Hosts_ guide.
```

I do not understad this change, we have the same in master too. Are all these links are supposed to link here:

```
[id="registering-a-host_managing-hosts"]
= Registering a Host to {ProjectName}
```

Comment (lzap) [reply guides/doc-Managing_Hosts/topics/Registering_Hosts.adoc:25]: Asked here: https://github.com/theforeman/foreman-documentation/pull/359/files#r714105415

**Comment by ryandeussing** [comment]
@lzap thanks, done!

**Comment by lzap** [comment]
What do you want to do with the xref link? Remove just this one? Remove all? Or fix them?

I asked in the PR which added it. I do not understand why it was added without id.

**Comment by lzap** [comment]
I fixed RHEL links in here: https://github.com/theforeman/foreman-documentation/commit/af69eddb2908dc3778841ae0ca89c4747176db58

* f6da3c7a..c845b6b8  3.0 -> 3.0
* 04ad8fff..4d218f63  2.5 -> 2.5

And the registration links here: https://github.com/theforeman/foreman-documentation/pull/727

* bb7f160..f6da3c7 3.0 -> 3.0
* 0e78618..04ad8ff 2.5 -> 2.5

Please verify, you should be fine now.

**Comment by ryandeussing** [comment]
Thank you @lzap

**Pull request closed without merging**